atlasdb-config/src/main/java/com/palantir/atlasdb/factory/PersistentStoragePathSanitizer.java

```diff
@@ -21,70 +21,63 @@
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.util.Comparator;
-import java.util.HashSet;
 import java.util.List;
-import java.util.Set;
-import java.util.regex.Pattern;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
-import com.google.common.base.MoreObjects;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 import com.palantir.logsafe.Preconditions;
 import com.palantir.logsafe.SafeArg;
 
-public final class PersistentStorageFactories {
-    private static final Pattern UUID_PATTERN = Pattern.compile(
-            "^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$");
-
-    private static final Set<String> SANITIZED_PATHS = new HashSet<>();
-
-    private PersistentStorageFactories() {}
+public final class PersistentStoragePathSanitizer {
+    private static final Logger log = LoggerFactory.getLogger(PersistentStoragePathSanitizer.class);
+    private static final String MAGIC_SUFFIX = "atlasdb-persistent-storage";
 
+    private PersistentStoragePathSanitizer() {}
 
     /**
-     * For the given path does the following: 1) it is sanitized only once per VM lifetime 2) if it exists checks that
-     * it is a directory, 3) if it is a directory removes all sub-folders whose names are string representation of a
-     * UUID.
+     * For the given path deletes all files and directories under the folder named {@code MAGIC_SUFFIX}.
      *
      * @param storagePath to the proposed storage location
      */
-    public static synchronized void sanitizeStoragePath(String storagePath) {
-        if (SANITIZED_PATHS.contains(storagePath)) {
-            return;
-        }
+    public static Path sanitizeStoragePath(String storagePath) {
+        File storageDirectory = new File(storagePath, MAGIC_SUFFIX);
+
+        Preconditions.checkArgument(
```

Comment:
what happens if `storagePath` doesn't exist, does this catch it?

Reply:
It is caught.